### ISSUE TITLE
bring up #19 (PSR-14) for revisting

### ISSUE THREAD

**Issue by HenryVolkmer** [doctrine/event-manager#64]
4 years are passed since [discussion about implementing PSR-14](https://github.com/doctrine/event-manager/issues/19), i want to bring up this topic again.

I propose to implement PSR-14: the grow in complexity is manageable and BC avoidable.
It would be an pleasure for me to hand out a PR for this.

**Comment by greg0ire** [comment]
> For now, it looks like PSR-14 supports stuff like stopping propagation, which doesn't look like a good fit for this component, as it complicates everything tenfold.

Feel free to work on a PR, and let's see if that comment holds true? It might help to discuss this around a PR showing what changes in practice.

**Comment by derrabus** [comment]
How and why would we implement PSR-14 on this library? What would be the gain?

**Comment by greg0ire** [comment]
Maybe I'm wrong but I thought we would implement PSR-14 in dependent libraries, and abandon this one. The gain would be one less dependency to maintain.

**Comment by HenryVolkmer** [comment]
> 

oh ok, i did not know.
My intention with this topic is to create interop with arbitrary PSR-14 EMs [here](https://github.com/doctrine/dbal/blob/3.5.x/src/DriverManager.php#L193). If this collides with project plans, then please nevermind and close this.

**Comment by greg0ire** [comment]
Let me rephrase: I thought your plan with this proposal would be to implement PSR-14 in `doctrine/dbal` and drop the dependency to `doctrine/event-manager`. I currently don't have such plans.

**Comment by greg0ire** [comment]
Why are you closing :sweat_smile: ?

**Comment by greg0ire** [comment]
I currently don't have such plans as in: I've not given PSR-14 any thought. I'm not against it! Please clarify your proposal.

**Comment by derrabus** [comment]
DBAL is sunsetting its event system entirely. DBAL 4 is not using the event manager anymore. So nothing to gain here.

**Comment by derrabus** [comment]
For the ORM, we could think about it, but as @greg0ire said, we'd better change the ORM to consume PSR-14 and kill the event manager than to make it compatible with PSR-14.

**Comment by HenryVolkmer** [comment]
I think, ORM consuming PSR-14 is the way to go.  If is help needed (I doubt it) then feel free and ping me.

**Comment by derrabus** [comment]
> If is help needed

Well, to me knowledge, nobody's actively working on that topic.
